diesel_migrations/src/file_based_migrations.rs

```diff
@@ -0,0 +1,308 @@
+        })
+    }
+
+    pub fn find_migrations_directory() -> Result<Self, MigrationError> {
+        Self::find_migrations_directory_in_path(&std::env::current_dir()?)
+    }
+
+    pub fn find_migrations_directory_in_path(
+        path: impl AsRef<Path>,
+    ) -> Result<Self, MigrationError> {
+        let migrations_directory = search_for_migrations_directory(path.as_ref())?;
+        Self::from_path(&migrations_directory)
+    }
+
+    #[doc(hidden)]
+    pub fn path(&self) -> &Path {
+        &self.base_path
+    }
+}
+
+fn search_for_migrations_directory(path: &Path) -> Result<PathBuf, MigrationError> {
+    migrations_internals::search_for_migrations_directory(path)
+        .ok_or_else(|| MigrationError::MigrationDirectoryNotFound(path.to_path_buf()))
+}
+
+fn migrations_directories<'a>(
+    path: &'a Path,
+) -> impl Iterator<Item = Result<DirEntry, MigrationError>> + 'a {
+    migrations_internals::migrations_directories(path).map(move |e| e.map_err(Into::into))
+}
+
+fn migrations_in_directory<'a, DB: Backend>(
+    path: &'a Path,
+) -> impl Iterator<Item = Result<SqlFileMigration, MigrationError>> + 'a {
+    migrations_directories(path).map(|entry| SqlFileMigration::from_path(&entry?.path()))
+}
+
+impl<DB: Backend> MigrationSource<DB> for FileBasedMigrations {
+    fn migrations(
+        &self,
+    ) -> Result<Vec<Box<dyn Migration<DB>>>, Box<dyn std::error::Error + Send + Sync + 'static>>
+    {
+        migrations_in_directory::<DB>(&self.base_path)
+            .map(|r| Ok(Box::new(r?) as Box<dyn Migration<DB>>))
+            .collect()
+    }
+}
+
+struct SqlFileMigration {
+    base_path: PathBuf,
+    metadata: TomlMetadataWrapper,
+    name: DieselMigrationName,
+}
+
+impl SqlFileMigration {
+    fn from_path(path: &Path) -> Result<Self, MigrationError> {
+        if migrations_internals::valid_sql_migration_directory(path) {
+            let metadata = TomlMetadataWrapper(
+                TomlMetadata::read_from_file(&path.join("metadata.toml")).unwrap_or_default(),
+            );
+            Ok(Self {
+                base_path: path.to_path_buf(),
+                metadata,
+                name: DieselMigrationName::from_path(path)?,
+            })
+        } else {
+            Err(MigrationError::UnknownMigrationFormat(path.to_path_buf()))
+        }
+    }
+}
+
+impl<DB: Backend> Migration<DB> for SqlFileMigration {
+    fn run(
+        &self,
+        conn: &dyn BoxableConnection<DB>,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+        Ok(run_sql_from_file(
+            conn,
+            &self.base_path.join("up.sql"),
+            &self.name,
+        )?)
+    }
+
+    fn revert(
+        &self,
+        conn: &dyn BoxableConnection<DB>,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+        Ok(run_sql_from_file(
+            conn,
+            &self.base_path.join("down.sql"),
+            &self.name,
+        )?)
+    }
```

Comment:
I think this would be all that's necessary to do reverts from within code, which would satisfy the use case of being able to run reverts from within docker. Thanks!

Reply:
You likely mean the corresponding method in `embeded_migrations.rs`, as this one is for file based migration which needs to exist at runtime. Or to word it differently: This method already exists in diesel 1.4.x.